### ISSUE TITLE
Switch on Qt5

### ISSUE THREAD

**Issue by glixx** [highperformancecoder/scidavis-old#117]
It's time for Qt5

**Comment by highperformancecoder** [comment]
On Mon, Jan 28, 2019 at 10:46:40PM -0800, glixx wrote:
> It's time for Qt5

Long overdue, in fact. However, this will require a version bump to
SciDAVis 2.x, and I'm also working on getting Python to work for
Windows, as well as the Qwt6 port into the same version bump.

I'm happier to release 2.x earlier if there's a strong demand for
it. It will mean a version bump to 3.x when the python changes and
Qwt6 go in, of course.

Cheers
-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders
Visiting Senior Research Fellow        hpcoder@hpcoders.com.au
Economics, Kingston University         http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Comment by alxpl** [comment]
I've just built the latest snapshot against Miquel's qwt5-qt5 in my [COPR](https://copr.fedorainfracloud.org/coprs/alexpl/scidavis/) for Fedora 28, 29 and rawhide. It seems to work fine and certainly the visual glitches present in Qt4 are gone. I'll try to update it weekly, but no guarantees.
Should I add a small news item in SourceForge, asking for (Fedora) user feedback?

**Comment by yeKcim** [comment]
If only I could help with code…
qt5 migration could also help for debian packaging (so ubuntu, mint,… integration).
♥ scidavis

**Comment by ronisbr** [comment]
Just to add to the discussion: openSUSE will very soon drop the support for Qt4, making it impossible to build and provide a package using the distribution system.

https://lists.opensuse.org/opensuse-factory/2019-03/msg00373.html

**Comment by highperformancecoder** [comment]
Thanks for the heads up. Hopefully, they will break it in Tumbleweed
first so we get some warning.

Note that SciDAVis doesn't build in in SLE15 because nothing provides
libqt4-devel.

The current codebase supports Qt5, but a) the SciDAVis python API
depends on the Qt version, so we need to do a version bump, and b) the
Qt5 version is not well tested.

However, before doing a version bump, I would also like to

a) move to Qwt6
b) move to Python3 (which is supported in the current codebase)
c) remove PyQt and sip from the dependencies, and replace with
boost.python and classdesc
d) replace the QtAssistant stuff with a call out to a web browser
loading the documentation, like what I do with Minsky.

I suspect a) will in part depend on c), which is well underway.

My preference is to have all of these changes in hand before bumping
the version. However, if necessary, we could aim for a short lived 2.x
series of releases, and postpone the big stuff for a 3.x release, which might
occur later this year.

Maybe I should do a beta 2.0.x release to at least give people the
chance to shake out any bugs should we need to do this.

Cheers

On Mon, Apr 01, 2019 at 11:57:55AM -0700, Ronan Arraes Jardim Chagas wrote:
> Just to add to the discussion: openSUSE will very soon drop the support for
> Qt4, making it impossible to build and provide a package using the distribution
> system.
> 
> https://lists.opensuse.org/opensuse-factory/2019-03/msg00373.html
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.*
> 

-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders
Visiting Senior Research Fellow        hpcoder@hpcoders.com.au
Economics, Kingston University         http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Comment by glixx** [comment]
Debian removed scidavis from repo already. So qt5 release is important as soon as possible to stop removing scidavis from repositories.

**Comment by highperformancecoder** [comment]
On Mon, Apr 01, 2019 at 06:23:13PM -0700, glixx wrote:
> Debian removed scidavis from repo already. So qt5 release is important as soon
> as possible to stop removing scidavis from repositories.

As far as I'm aware, SciDAVis is only available from my personal
OpenSUSE build service repository
(https://build.opensuse.org/project/show/home:hpcoder1). I don't have
resources to try and get it into official repos (it bad enough getting
it to build on OBS). Any versions available in other repos tend to be
out of date from what I've seen.

SciDAVis is still very much available for the latest Debian from there (Debian
9). As mentioned above, the only distro it is not available for is the
latest SuSE enterprise (SLE15).

Cheers


-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders
Visiting Senior Research Fellow        hpcoder@hpcoders.com.au
Economics, Kingston University         http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Comment by ronisbr** [comment]
Until weeks ago Scidavis was available officially in Tumbleweed. It broke because a Qt4 package was dropped.

**Comment by highperformancecoder** [comment]
On Mon, Apr 01, 2019 at 08:55:08PM -0700, Ronan Arraes Jardim Chagas wrote:
> Until weeks ago Scidavis was available officially in Tumbleweed. It broke
> because a Qt4 package was dropped.

SciDAVis is still available. I just installed it then on my system,
which was dupdated last Friday. It also installs on the official
docker Tumbleweed image.

But yes, I agree it could be going away soon...

I'll make a beta package so that bleeding edgers can try it
out... When I have a day spare, that is, as it will take that long :).

-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders
Visiting Senior Research Fellow        hpcoder@hpcoders.com.au
Economics, Kingston University         http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Comment by krop** [comment]
fwiw,
scidavis 1.25 request: https://build.opensuse.org/request/show/698903
qwtplot3d-qt5 request: https://build.opensuse.org/request/show/698902
qwt5-qt5 request: https://build.opensuse.org/request/show/698687

(edited to clarify my comment)
From my tests, it looks bad with dark color schemes when using either Qt4 or Qt5.

I agree with @highperformancecoder however, packaging it is a pain due tu qwt and qwtplot3d to be mostly abandoned by their authors.

**Comment by highperformancecoder** [comment]
I have just come back from a months travelling, and can confirm that SciDAVis is now broken on all RPM repos, due to a breaking change in the qwt spec file. I have now created a libqwt5-qt4 project, which builds on some, though not all distros, however, I'm now in the process of waiting several days before the artifacts are available for building downstream projects like scidavis.

Once this is done, its a case of rinse and repeat to set up scidavis 2.0.x beta using qt5. I'm going to be very busy from next week with another project, so arfifact delays will probably end up blowing this whole thing out by months! Gnaarrgghh

**Comment by archisman-panigrahi** [comment]
The new version of SciDAVis with qt5 seems to work well in KDE Neon. @highperformancecoder Please submit it to the official Debian repositories once again.

**Comment by highperformancecoder** [comment]
On Sat, May 02, 2020 at 02:41:42AM -0700, Archisman Panigrahi wrote:
> The new version of SciDAVis with qt5 seems to work well in KDE Neon.
> @highperformancecoder Please submit it to official Debian repositories once
> again.

I don't know how to do that :).

-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders     hpcoder@hpcoders.com.au
                      http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Comment by highperformancecoder** [comment]
With the 2.1 release of SciDAVis, the world of python2 and Qt4 have been left behind.
